clang/lib/CodeGen/CodeGenFunction.h

```diff
@@ -4987,7 +4990,11 @@ class CodeGenFunction : public CodeGenTypeCache {
                                      const llvm::Twine &Name = "");
   // Adds a convergence_ctrl token to |Input| and emits the required parent
   // convergence instructions.
-  llvm::CallBase *addControlledConvergenceToken(llvm::CallBase *Input);
+  template <typename CallType>
+  CallType *addControlledConvergenceToken(CallType *Input) {
+    return dyn_cast<CallType>(
```

Comment:
should just be cast? You also have yet another dyn_cast on the use?

Reply:
Forgot to fix the use, and yes, should be a cast. Bad habits...